IDEAS/Controls/ControlHeating/RunningMeanTemperatureEN15251.mo

```diff
@@ -2,39 +2,32 @@ within IDEAS.Controls.ControlHeating;
 model RunningMeanTemperatureEN15251
   "Calculate the running mean temperature of 7 days, acccording to norm EN15251"
 
-  parameter Real[7] TAveDayIni(unit="K", displayUnit="degC", fixed=false)
-    "Initial running mean temperature";
-
-  // Interface
    discrete Modelica.Blocks.Interfaces.RealOutput TRm(unit="K",displayUnit = "degC")
     "Running mean average temperature"
      annotation (Placement(transformation(extent={{96,-10},{116,10}})));
+  Modelica.Blocks.Sources.RealExpression TAmb(y=sim.Te)
+    annotation (Placement(transformation(extent={{-80,-10},{-60,10}})));
+  outer BoundaryConditions.SimInfoManager sim
+    annotation (Placement(transformation(extent={{-100,80},{-80,100}})));
 
 protected
-  discrete Real[7] TAveDay(unit="K",displayUnit = "degC")
-    "Vector with the average day temperatures of the previous nTermRm days";
+  parameter Modelica.SIunits.Time t_start(fixed=false) "Start time of the model";
   parameter Real coeTRm[7] = {1, 0.8, 0.6, 0.5, 0.4, 0.3, 0.2}./3.8
     "weighTAmb.yg coefficient for the running average";
-
+  discrete Real[7] TAveDay(each unit="K",each displayUnit = "degC")
+    "Vector with the average day temperatures of the previous nTermRm days";
   Real intTAmb "integral of TAmb.y";
-  parameter Modelica.SIunits.Time t_start(fixed=false) "Start time of the model";
 
-public
-  Modelica.Blocks.Sources.RealExpression TAmb(y=sim.Te)
-    annotation (Placement(transformation(extent={{-80,-10},{-60,10}})));
-  outer BoundaryConditions.SimInfoManager sim
-    annotation (Placement(transformation(extent={{-100,80},{-80,100}})));
+
 initial equation
-      t_start = time;
-      TAveDayIni = ones(7).*sim.Te;
+  intTAmb=0;
+  t_start = time;
+  TAveDay=ones(7).*sim.Te;
+  TRm=sim.Te;
 equation
   der(intTAmb) =  TAmb.y;
 algorithm
-  when initial() then
-    // initialization of the discrete variables
-    TAveDay:=TAveDayIni;
-    TRm:=TAveDayIni[1];
-  elsewhen sample(t_start+24*3600,24*3600) then
+  when sample(t_start+24*3600,24*3600) then
```

Comment:
initial equation should thus be preferred to `when initial()`?

Reply:
That solved the warning but I think that both approaches should give the same result.